### DISCUSSION TITLE
How to set formState manully?

### DISCUSSION THREAD

**Discussion by michael-land** [react-hook-form/react-hook-form#4052] [General] [answered]
My use case would be allow user restore data from previous input. 

1. user type something
2. user refresh the page, the form component mounted ( now the form state is clean and isDirty is false )
3. ask user if he want to restore previous data
4. if yes, restore the data use `form.reset` ( the form is not dirty, because user didn't changed the form )

The problem is that the user is not able to submit the form immediately unless they change something. 

P.S. the submit button is disabled when form is not dirty or is not submitting

Is there any way to set formState manually or could it be add in V7?
  

```tsx
// part of recover logic
<Button
  onClick={() => {
    form.reset(JSON.parse(localStorage.getItem(key) ?? '{}') as any);
    toast('Data restored');
  }}
>
  Recover
</Button>
```

**Comment by bluebill1049** [comment, marked as answer]
sounds like you will have to do two reset?
```
reset(defaultValues);
```

```
reset(data, { // updated data
  keepDefaultValues: true, // v7
})
```

**Reply by michael-land** [reply]
Thanks, I managed solved it with `form.control.updateFormState({ isDirty: true });` in v6.

**Reply by bluebill1049** [reply]
careful with control `object`, it's not safe to use. in V7 its been clean up a lot in that object.
